test/apps/iodemo/io_demo.cc

```diff
@@ -172,21 +250,75 @@ class IoDemoRandom {
         }
     }
 
-    static inline void fill(unsigned &seed, void *buffer, size_t size) {
+    static void *get_host_fill_buffer(void *buffer, size_t size,
+                                      ucs_memory_type_t memory_type)
+    {
+#ifdef HAVE_CUDA
```

Comment:
why need ifdef?

Reply:
Because there's a cal to `cudaMemcpy()` inside - this symbol doesn't exist at all when `HAVE_CUDA` is not defined.

Reply:
i don't see a call cudaMemcpy inside..